test/conformance/exp_command_buffer/buffer_fill_kernel_update.cpp

```diff
@@ -0,0 +1,404 @@
+    char zero = 0;
+    ASSERT_SUCCESS(urEnqueueMemBufferFill(queue, new_buffer, &zero,
+                                          sizeof(zero), 0, buffer_size, 0,
+                                          nullptr, nullptr));
+    ASSERT_SUCCESS(urQueueFinish(queue));
+
+    // Set argument index zero as new buffer
+    ur_exp_command_buffer_update_memobj_arg_desc_t new_output_desc = {
+        UR_STRUCTURE_TYPE_EXP_COMMAND_BUFFER_UPDATE_MEMOBJ_ARG_DESC, // stype
+        nullptr,                                                     // pNext
+        0,                                                           // argIndex
+        nullptr,    // pProperties
+        new_buffer, // hArgValue
+    };
+
+    // Set argument index 2 as new value to fill (index 1 is buffer accessor)
+    uint32_t new_val = 33;
+    ur_exp_command_buffer_update_value_arg_desc_t new_input_desc = {
+        UR_STRUCTURE_TYPE_EXP_COMMAND_BUFFER_UPDATE_VALUE_ARG_DESC, // stype
+        nullptr,                                                    // pNext
+        2,                                                          // argIndex
+        sizeof(new_val),                                            // argSize
+        nullptr,  // pProperties
+        &new_val, // hArgValue
+    };
+
+    ur_exp_command_buffer_update_kernel_launch_desc_t update_desc = {
+        UR_STRUCTURE_TYPE_EXP_COMMAND_BUFFER_UPDATE_KERNEL_LAUNCH_DESC, // stype
+        nullptr,                                                        // pNext
+        1,                // numMemobjArgs
+        0,                // numPointerArgs
+        1,                // numValueArgs
+        0,                // numExecInfos
+        0,                // workDim
+        &new_output_desc, // pArgMemobjList
+        nullptr,          // pArgPointerList
+        &new_input_desc,  // pArgValueList
+        nullptr,          // pArgExecInfoList
+        nullptr,          // pGlobalWorkOffset
+        nullptr,          // pGlobalWorkSize
+        nullptr,          // pLocalWorkSize
+    };
+
+    // Update kernel and enqueue command-buffer again
+    ASSERT_SUCCESS(
+        urCommandBufferUpdateKernelLaunchExp(command_handle, &update_desc));
+    ASSERT_SUCCESS(urCommandBufferEnqueueExp(updatable_cmd_buf_handle, queue, 0,
+                                             nullptr, nullptr));
+    ASSERT_SUCCESS(urQueueFinish(queue));
+
+    // Verify that update occurred correctly
+    ValidateBuffer(new_buffer, buffer_size, new_val);
+}
+
+// Test updating the global size so that the fill outputs to a larger buffer
+TEST_P(BufferFillCommandTest, UpdateGlobalSize) {
+    ASSERT_SUCCESS(urCommandBufferEnqueueExp(updatable_cmd_buf_handle, queue, 0,
+                                             nullptr, nullptr));
+    ASSERT_SUCCESS(urQueueFinish(queue));
+    ValidateBuffer(buffer, sizeof(val) * global_size, val);
+
+    size_t new_global_size = 64;
+    const size_t buffer_size = sizeof(val) * new_global_size;
```

Comment:
can we make this a new variable name? it looks like this is shadowing an existing declaration from the [fixture](https://github.com/oneapi-src/unified-runtime/pull/1089/files#diff-9fd0d900f6d64a39d5d438eec69265eeadd6d5c62e5d5815e4f259c920e6d8e3R49) (also applies to the `SeparateUpdateCalls` test)